### PR TITLE
Bench: use a normal package for the test suite

### PR DESCRIPTION
**Pull request by SkySkimmer**
instead of installing `coq --with-test`

Needs
- https://github.com/coq/opam/pull/3019

### CODE REVIEW

**Comment by SkySkimmer** [comment]
Test run https://gitlab.inria.fr/coq/coq/-/jobs/4261408

File: dev/bench/bench.sh [hunk 1]
```diff
@@ -70,17 +70,16 @@ check_variable () {
...
 : "${new_coq_repository:=$CI_REPOSITORY_URL}"
 : "${old_coq_repository:=$CI_REPOSITORY_URL}"
-: "${new_coq_opam_archive_git_uri:=https://github.com/coq/opam-coq-archive.git}"
-: "${old_coq_opam_archive_git_uri:=https://github.com/coq/opam-coq-archive.git}"
-: "${new_coq_opam_archive_git_branch:=master}"
-: "${old_coq_opam_archive_git_branch:=master}"
+: "${new_coq_opam_archive_git_uri:=https://github.com/skyskimmer/opam-coq-archive.git}"
+: "${old_coq_opam_archive_git_uri:=https://github.com/skyskimmer/opam-coq-archive.git}"
+: "${new_coq_opam_archive_git_branch:=coq-test}"
+: "${old_coq_opam_archive_git_branch:=coq-test}"
```
Comment (SkySkimmer) [inline dev/bench/bench.sh:76]: Note: undo this change once testing is done

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by SkySkimmer** [comment]
Test with fixed test suite https://gitlab.inria.fr/coq/coq/-/jobs/4261929 (OLD = NEW = this PR)

**Comment by SkySkimmer** [comment]
It works

**Comment by SkySkimmer** [comment]
@coq/bench-maintainers this should be ready

**Comment by SkySkimmer** [comment]
ping @coq/bench-maintainers

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now
